datafusion/core/tests/sqllogictests/test_files/groupby.slt

```diff
@@ -1921,3 +1921,96 @@ SELECT DISTINCT + col1 FROM tab2 AS cor0 GROUP BY cor0.col1
 41
 59
 61
+
+
+
+# Columns in the table are a,b,c,d. Source is CsvExec which is ordered by
+# a,b,c column. Column a has cardinality 2, column b has cardinality 4.
+# Column c has cardinality 100 (unique entries). Column d has cardinality 5.
+statement ok
+CREATE EXTERNAL TABLE annotated_data_finite2 (
+  a0 INTEGER,
+  a INTEGER,
+  b INTEGER,
+  c INTEGER,
+  d INTEGER
+)
+STORED AS CSV
+WITH HEADER ROW
+WITH ORDER (a ASC, b ASC, c ASC)
+LOCATION 'tests/data/window_2.csv';
+
```

Comment:
In the current version 
```
RepartitionExec: partitioning=Hash([Column { name: "b", index: 0 }, Column { name: "a", index: 1 }], 4), input_partitions=4
  RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1 
```
stack messes ordering. Hence to not mess up ordering, when source is infinite. We should run query with single partition. Hence, we should add here
```sql
statement ok
set datafusion.execution.target_partitions = 1;
```
also.